### PR TITLE
linter: add skywalking-eyes into bazel

### LINKED ISSUE #40789: add license check into bazel and run it by bazel
## Enhancement

Now we run the license check in the GitHub action. but we always overhead its limit. so we need to migrate it from GHA to jenkins.

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #40789

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

If you remove the license header. CI will fail.
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by crazycs520 — APPROVED** [review]

File: build/nogo_config.json [hunk 1]
```diff
@@ -305,7 +305,7 @@
   "misspell": {
     "exclude_files": {
       "/cgo/": "ignore cgo code",
-      "/external/": "no need to vet third party code",
+      "external/": "no need to vet third party code",
```
Comment (Defined2014) [inline build/nogo_config.json:308]: why change it?

Comment (hawkingrei) [reply build/nogo_config.json:308]: misspell will find the misspell problem in the ```skywalking-eyes```. We need to skip it. the old config has a problem.

**Review by Defined2014 — APPROVED** [review]

File: build/nogo_config.json (hunk 1 above)
Comment (hawkingrei) [reply build/nogo_config.json:308]: fix all the lint config.

**Pull request merged**